include/sea_dsa/Graph.hh

```diff
@@ -134,110 +153,128 @@ public:
+  /// present
+  virtual Cell &mkCell(const llvm::Value &v, const Cell &c);
+  virtual Cell &mkRetCell(const llvm::Function &fn, const Cell &c);
 
-    /// return a cell for the value
-    virtual const Cell &getCell(const llvm::Value &v);
+  /// return a cell for the value
+  virtual const Cell &getCell(const llvm::Value &v);
 
-    /// return true iff the value has a cel
-    virtual bool hasCell(const llvm::Value &v) const;
+  /// return true iff the value has a cel
+  virtual bool hasCell(const llvm::Value &v) const;
 
-    virtual bool hasScalarCell(const llvm::Value &v) {
-      return m_values.count(&v) > 0;
-    }
+  virtual bool hasScalarCell(const llvm::Value &v) {
+    return m_values.count(&v) > 0;
+  }
 
-    virtual bool hasRetCell(const llvm::Function &fn) const {
-      return m_returns.count(&fn) > 0;
-    }
+  virtual bool hasRetCell(const llvm::Function &fn) const {
+    return m_returns.count(&fn) > 0;
+  }
 
-    virtual Cell &getRetCell(const llvm::Function &fn);
+  virtual Cell &getRetCell(const llvm::Function &fn);
 
-    virtual const Cell &getRetCell(const llvm::Function &fn) const;
+  virtual const Cell &getRetCell(const llvm::Function &fn) const;
 
-    llvm::Optional<DsaAllocSite *> getAllocSite(const llvm::Value &v) const {
-      auto it = m_valueToAllocSite.find(&v);
-      if (it != m_valueToAllocSite.end())
-        return it->second;
+  llvm::Optional<DsaAllocSite *> getAllocSite(const llvm::Value &v) const {
+    auto it = m_valueToAllocSite.find(&v);
+    if (it != m_valueToAllocSite.end())
+      return it->second;
 
-      return llvm::None;
-    }
+    return llvm::None;
+  }
 
-    DsaAllocSite *mkAllocSite(const llvm::Value &v);
+  DsaAllocSite *mkAllocSite(const llvm::Value &v);
 
-    llvm::iterator_range<alloc_site_iterator> alloc_sites() {
-      alloc_site_iterator begin = m_allocSites.begin();
-      alloc_site_iterator end = m_allocSites.end();
-      return llvm::make_range(begin, end);
-    }
+  llvm::iterator_range<alloc_site_iterator> alloc_sites() {
+    alloc_site_iterator begin = m_allocSites.begin();
+    alloc_site_iterator end = m_allocSites.end();
+    return llvm::make_range(begin, end);
+  }
 
-    llvm::iterator_range<alloc_site_const_iterator> alloc_sites() const {
-      alloc_site_const_iterator begin = m_allocSites.begin();
-      alloc_site_const_iterator end = m_allocSites.end();
-      return llvm::make_range(begin, end);
-    }
+  llvm::iterator_range<alloc_site_const_iterator> alloc_sites() const {
+    alloc_site_const_iterator begin = m_allocSites.begin();
+    alloc_site_const_iterator end = m_allocSites.end();
+    return llvm::make_range(begin, end);
+  }
 
-    bool hasAllocSiteForValue(const llvm::Value &v) const {
-      return m_valueToAllocSite.count(&v) > 0;
-    }
+  bool hasAllocSiteForValue(const llvm::Value &v) const {
+    return m_valueToAllocSite.count(&v) > 0;
+  }
+
+  llvm::Optional<DsaCallSite *> getCallSite(const llvm::Value &v) const {
```

Comment:
kinda strange that call site is looked up for a value. Would be more appropriate to resolve `llvm::CallSite` to `DsaCallSite`. This avoid confusion that many different values look like they might be a call site when in fact they cannot